# 2° Preentrega - Curso Data Science - Ingenias 2024

#### Tema de entrega
- Seleccionar dataset grupal
- Describir el objetivo del trabajo grupal
- Realizar exploraciones y transformaciones con pandas
- Visualizaciones con pandas

## Análisis de bullying en colegios secundarios de Argentina

Se analiza un cuestionario autoadministrado de 72 preguntas a alumnos de entre 11 y 18 años de edad. 
Engloba los siguientes aspectos:
- edad
- género
- alimentación
- lesiones graves
- intimidación
- seguridad vial
- amistades
- tabaquismo
- alcoholismo
- drogas
- sexualidad
- actividad física
- tiempo libre
- vinculo familiar

### Objetivo
El objetivo del presente trabajo es analizar si existen correlaciones entre bullying (intimidación) y cada una de las variables anteriormente mencionadas.

In [22]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt

# Los siguientes comentarios son acerca de configuraciones previas para el entorno de Python en cuanto a display

# This sets no limits
#pd.set_option('display.max_columns', None)
#pd.set_option('display.max_rows', None)

# This resets pandas display config
pd.reset_option('^display.', silent=True)

# This resets NumPy display config
np.set_printoptions(edgeitems=3, infstr='inf',
                   linewidth=75, nanstr='nan', precision=8,
                   suppress=False, threshold=1000, formatter=None)

In [10]:
bullying = pd.read_csv("EMSE_DatosAbiertos.csv")

In [11]:
bullying.info

<bound method DataFrame.info of       sitio  record   q1 texto_q1   q2   texto_q2   q3  \
0        AG       1  3.0  13 años  2.0   Femenino  2.0   
1        AG       2  3.0  13 años  2.0   Femenino  2.0   
2        AG       3  4.0  14 años  1.0  Masculino  4.0   
3        AG       4  6.0  16 años  1.0  Masculino  2.0   
4        AG       5  3.0  13 años  2.0   Femenino  2.0   
...     ...     ...  ...      ...  ...        ...  ...   
56976    AG   57091  6.0  16 años  1.0  Masculino  5.0   
56977    AG   57092  6.0  16 años  1.0  Masculino  5.0   
56978    AG   57093  6.0  16 años  1.0  Masculino  NaN   
56979    AG   57094  6.0  16 años  2.0   Femenino  5.0   
56980    AG   57095  6.0  16 años  2.0   Femenino  5.0   

                                                texto_q3    q4    q5  ...  \
0      9no grado nivel Primario/Polimodal o 2do año n...   NaN   NaN  ...   
1      9no grado nivel Primario/Polimodal o 2do año n...   NaN   NaN  ...   
2      2do año/11vo grado nivel Polimoda

In [12]:
bullying.describe()

record            q1            q2            q3            q4  \
count  56981.000000  56873.000000  56445.000000  55920.000000  36052.000000   
mean   28535.389832      4.905403      1.520188      3.064485      1.642857   
std    16479.971553      1.358115      0.499597      1.324451      0.096898   
min        1.000000      1.000000      1.000000      1.000000      1.200000   
25%    14263.000000      4.000000      1.000000      2.000000      1.570000   
50%    28522.000000      5.000000      2.000000      3.000000      1.640000   
75%    42804.000000      6.000000      2.000000      4.000000      1.700000   
max    57095.000000      8.000000      2.000000      5.000000      2.030000   

                 q5            q6           q10           q15           q16  \
count  36052.000000  56482.000000  56771.000000  56741.000000  56713.000000   
mean      60.212193      1.460448      1.802734      1.361573      1.496447   
std       14.723909      0.764121      1.451475      1.079432      1.211089   
min       27.000000      1.000000      1.000000      1.000000      1.000000   
25%       51.000000      1.000000      1.000000      1.000000      1.000000   
50%       58.000000      1.000000      1.000000      1.000000      1.000000   
75%       66.000000      2.000000      2.000000      1.000000      1.000000   
max      187.000000      5.000000      8.000000      8.000000      8.000000   

       ...      qnnotb2g      qnbcanyg         qnc1g        qnpa5g  \
count  ...  55908.000000  19735.000000  19696.000000  55551.000000   
mean   ...      1.208968      1.089891      1.330575      1.701932   
std    ...      0.406575      0.286033      0.470432      0.457414   
min    ...      1.000000      1.000000      1.000000      1.000000   
25%    ...      1.000000      1.000000      1.000000      1.000000   
50%    ...      1.000000      1.000000      1.000000      2.000000   
75%    ...      1.000000      1.000000      2.000000      2.000000   
max    ...      2.000000      2.000000      2.000000      2.000000   

             qnpa7g        qnpe3g        qnpe5g        weight       stratum  \
count  55551.000000  55032.000000  55032.000000  56981.000000  5.698100e+04   
mean       1.828428      1.633431      1.687237     46.288166  2.018129e+08   
std        0.377011      0.481872      0.463623     80.162741  7.068990e+03   
min        1.000000      1.000000      1.000000      2.120000  2.018010e+08   
25%        2.000000      1.000000      1.000000     14.420000  2.018070e+08   
50%        2.000000      2.000000      2.000000     26.370000  2.018130e+08   
75%        2.000000      2.000000      2.000000     44.780000  2.018200e+08   
max        2.000000      2.000000      2.000000   1192.430000  2.018240e+08   

                psu  
count  56981.000000  
mean      12.438234  
std        9.378964  
min        1.000000  
25%        6.000000  
50%       11.000000  
75%       17.000000  
max       57.000000  

[8 rows x 157 columns]

In [13]:
bullying.shape

(56981, 309)

In [25]:
cols=bullying.columns
print(cols)

Index(['sitio', 'record', 'q1', 'texto_q1', 'q2', 'texto_q2', 'q3', 'texto_q3',
       'q4', 'q5',
       ...
       'texto_qnpa5g', 'qnpa7g', 'texto_qnpa7g', 'qnpe3g', 'texto_qnpe3g',
       'qnpe5g', 'texto_qnpe5g', 'weight', 'stratum', 'psu'],
      dtype='object', length=309)


Estudio de los valores que toma cada columna:

In [17]:
diccionario_tabla = {"Columna":[],"Valores":[]}

for col in cols:
    diccionario_tabla["Columna"].append(col)
    diccionario_tabla["Valores"].append(bullying[col].unique())
    
dataframe_tabla = pd.DataFrame(diccionario_tabla)
print(dataframe_tabla)

            Columna                                            Valores
0             sitio                                               [AG]
1            record  [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14...
2                q1      [3.0, 4.0, 6.0, 2.0, 5.0, 1.0, 7.0, nan, 8.0]
3          texto_q1  [13 años, 14 años, 16 años, 12 años, 15 años, ...
4                q2                                    [2.0, 1.0, nan]
5          texto_q2                [Femenino, Masculino, Dato perdido]
6                q3                     [2.0, 4.0, nan, 3.0, 5.0, 1.0]
7          texto_q3  [9no grado nivel Primario/Polimodal o 2do año ...
8                q4  [nan, 1.67, 1.73, 1.72, 1.64, 1.5, 1.62, 1.66,...
9                q5  [nan, 56.0, 70.0, 48.0, 45.0, 40.0, 52.0, 92.0...
10               q6                     [2.0, 1.0, 3.0, 4.0, 5.0, nan]
11         texto_q6  [Rara vez, Nunca, Algunas veces, Casi siempre,...
12              q10      [1.0, 4.0, 3.0, 6.0, 5.0, 2.0, nan, 7.0, 8.0]
13    

En nuestro dataset hay muchas columnas (309) y casi todas duplicadas, ya que para cada respuesta obtenida, al string ya fue asignada una variable categórica antes de nuestro análisis.

Eliminamos las columnas que tengan variables numéricas a partir de strings, y nos quedamos con la variable string.

Ejemplo: de las columnas q1 y texto_q1 nos quedamos solo con texto_q1 y eliminamos q1

In [24]:
import re

columnas_a_eliminar=[]
for col in cols:
    match = re.match(r'^texto_q(\d+)$', col)
    if match:
        numero = match.group(1)
        q_col = f'{numero}'
        if q_col in cols:
            columnas_a_eliminar.append(q_col)

# Eliminar las columnas identificadas
bullying.drop(columns=columnas_a_eliminar, inplace=True)

print(columnas_a_eliminar)

[]
